src/utils/safeHref.ts

```diff
@@ -21,33 +31,38 @@ export function configureSafeHref(configObject: SafeHrefConfigObj) {
     if (configObject.origin) {
         globalOrigin = configObject.origin
     }
+
+    if (configObject.additionalProtocols && configObject.additionalProtocols.length) {
+        customProtocols.push(...configObject.additionalProtocols)
+    }
 }
 
 export function getUseSafeHref(): boolean {
     return useSafeHref
 }
 
-export function getURLInfo(url: string): URLInfo {
-    /**
-     * An array of the safely allowed url protocols
-     */
-    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:', 'data:']
+export function resetCutomProtocols() {
+    customProtocols = []
+}
 
+export function getURLInfo(url: string, allowProtocols: Array<string>): URLInfo {
     /**
      * - Find protocol of URL or set to 'relative'
      * - Find origin of URL
      * - Determine if sameOrigin
      * - Determine if protocol of URL is safe
      */
-    const protocolResult = url.match(PROTOCOL_REGEX)
-    const originResult = url.match(ORIGIN_REGEX)
+    const cleanedUrl = url.replace(WHITE_SPACE_REGEX, '')
```

Comment:
Why not `url.trim()`?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim

Reply:
I thought about that, for some reason in my head I didn't want to trim whitespace from the end of a url. But thinking about that all the way through, that doesn't make sense. Will adjust and switch over to trim